test/java/seedu/addressbook/person/NameIsSimilarTest.java

```diff
@@ -0,0 +1,66 @@
+package seedu.addressbook.person;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Name;
+
+public class NameIsSimilarTest {
+	
+	@Test
+	public void test1() throws IllegalValueException {
+		Name p1 = new Name("John");
+		Name p2 = new Name("John");
+	
+		assertTrue(p2.isSimilar(p1));
+	}
+	
+	@Test
+	public void test2() throws IllegalValueException {
+		Name p1 = new Name("John");
+		Name p2 = new Name("Saufi");
+	
+		assertFalse(p1.isSimilar(p2));
+	}
+	
+	@Test
+	public void test3() throws IllegalValueException {
+		Name p1 = new Name("John");
+		Name p2 = new Name("john");
+	
+		assertTrue(p1.isSimilar(p2));
+	}
+	
+	@Test
+	public void test4() throws IllegalValueException {
+		Name p1 = new Name("John");
+	
+		assertFalse(p1.isSimilar(null));
+	}
+	
+	@Test
+	public void test5() throws IllegalValueException {
+		Name p1 = new Name("John Smith");
+		Name p2 = new Name("smith john");
+	
+		assertTrue(p1.isSimilar(p2));
+	}
+	
+	@Test
+	public void test6() throws IllegalValueException {
+		Name p1 = new Name("John Smith");
+		Name p2 = new Name("smith");
+	
+		assertTrue(p1.isSimilar(p2));
+	}
+	
+	@Test
+	public void test7() throws IllegalValueException {
+		Name p1 = new Name("John Smith");
+		Name p2 = new Name(" John");
+	
```

Comment:
you can consider moving all the steps necessary for the test to run to a setup method before beginning the tests. It will reduce the number of steps